amt/services/storage.py

```diff
@@ -42,25 +42,54 @@ def init(storage_type: str = "file", **kwargs: Unpack[WriterFactoryArguments]) -
 class FileSystemStorageService(Storage):
     def __init__(self, location: Path = Path("./tests/data"), filename: str = "system_card.yaml") -> None:
         self.base_dir = location
-        if not filename.endswith(".yaml"):
-            raise ValueError(f"Filename {filename} must end with .yaml instead of .{filename.split('.')[-1]}")
         self.filename = filename
-        self.path = self.base_dir / self.filename
 
     def write(self, data: dict[str, Any]) -> None:
         if not Path(self.base_dir).exists():
             Path(self.base_dir).mkdir()
         with open(self.path, "w") as f:
             dump(data, f, default_flow_style=False, sort_keys=False)
 
-    def read(self) -> Any:  # noqa
-        # todo: this probably has to be moved to 'global scope'
-        yaml.add_constructor("!include", yaml_include.Constructor(base_dir=self.base_dir))
+    def read(self, autoload: bool = True) -> Any:  # noqa
+        yaml.add_constructor("!include", yaml_include.Constructor(base_dir=self.base_dir, autoload=autoload))
         with open(self.path) as f:
-            return yaml.full_load(f)
+            data = yaml.full_load(f)
+        del yaml.Loader.yaml_constructors["!include"]
+        del yaml.FullLoader.yaml_constructors["!include"]
+        return data
 
     def close(self) -> None:
         """
         This method is empty because with the `with` statement in the writer, Python will already close the writer
         after usage.
         """
+
+    @property
+    def base_dir(self) -> Path:
+        return self._base_dir
+
+    @base_dir.setter
+    def base_dir(self, location: Path) -> None:
+        self._base_dir = location
+
+    @property
+    def autoload(self) -> bool:
+        return self._autoload
+
+    @autoload.setter
+    def autoload(self, autoload: bool) -> None:
+        self._autoload = autoload
+
+    @property
+    def filename(self) -> str:
+        return self._filename
+
+    @filename.setter
+    def filename(self, filename: str) -> None:
+        if not filename.endswith(".yaml"):
+            raise ValueError(f"Filename {filename} must end with .yaml instead of .{filename.split('.')[-1]}")
+        self._filename = filename
+
+    @property
+    def path(self) -> Path:
+        return self._base_dir / self.filename
```

Comment:
I am not sure if we should be able to 'set' properties on the FileSystemStorageService, though it depends on how we intend to use it.

It seems to be set up with the idea that each FileSystemStorageService is made for 1 file, which you then can read and write. If one can change the "file we are working on" or the base_dir.. we risk reading/writing a different file than expected. If you don't really need the properties (and I assume we don't), I'd remove them.